### PR TITLE
libunistring 1.3

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by carlocab** [comment]
After downloading the built bottles here, running `brew test guile` fails. However, running the commands from the `guile` test manually succeeds, which is weird.

@Bo98 any ideas?

**Comment by carlocab** [comment]
Hmm, I see. The Sequoia builds are missing `-liconv` for some reason. Weird.

**Comment by Bo98** [comment]
Can have a look further at some point in the weekend but we should probably have a test for iconv support in libunistring directly.

**Comment by carlocab** [comment]
Probably at least in part covered by `make check`, but we skip it now for newer macOS.

File: Formula/lib/libunistring.rb [hunk 1]
```diff
@@ -22,6 +22,7 @@ def install
     system "./configure", "--disable-dependency-tracking",
                           "--disable-silent-rules",
                           "--prefix=#{prefix}"
+    ENV.append "LDFLAGS", "-liconv" if OS.mac? && MacOS.version >= :sequoia
```
Comment (carlocab) [inline Formula/lib/libunistring.rb:25]: Didn't help. Still no linkage to `libiconv`.

**Comment by Bo98** [comment]
Ahh might have to adjust this to be `>=`: https://github.com/Homebrew/brew/blob/6f17b0644771e0a1d7c15ac72471dae2ef87314a/Library/Homebrew/extend/os/mac/extend/ENV/super.rb#L137

File: Formula/lib/libunistring.rb [hunk 2]
```diff
@@ -18,12 +18,18 @@ class Libunistring < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "416da7b6b8d158de4adccc0479df3b8cc6a532f39f71df928dd979b01812da21"
   end
 
+  on_sonoma :or_newer do
+    # gnulib rejects new macOS iconv implementation.
+    # https://github.com/coreutils/gnulib/commit/bab130878fe57086921fa7024d328341758ed453
+    depends_on "libiconv"
+  end
```
Comment (carlocab) [inline Formula/lib/libunistring.rb:25]: Probably could be bad for dependents. Let's see...

**Comment by Bo98** [comment]
Seems very messy if it bleeds into gettext.

It has to be said, the situation has been like this for an entire year and precisely zero people appear to have noticed or complained.

**Comment by carlocab** [comment]
Bleeding into `gettext` is kinda inevitable. But this was just an experiment to see how testing went -- will change this to use `am_cv_func_iconv_works` instead.

**Comment by carlocab** [comment]
> It has to be said, the situation has been like this for an entire year and precisely zero people appear to have noticed or complained.

Note: it is not quite the case that precisely zero people have noticed or complained: https://github.com/Homebrew/homebrew-core/pull/142490#issuecomment-1793862125

**Review by carlocab — APPROVED** [review]

**Comment by ychin** [comment]
> Seems very messy if it bleeds into gettext.
> 
> It has to be said, the situation has been like this for an entire year and precisely zero people appear to have noticed or complained.

Per the other comment said, it's definitely not the case that zero people noticed! Eventually though I thought Apple fixed iconv around macOS 14.2 but from the [linked commit in gnulib](https://github.com/coreutils/gnulib/commit/bab130878fe57086921fa7024d328341758ed453) it specifically says macOS 14.4 is broken so perhaps there were more bugs that weren't fixed. 

Around macOS 14.2 Apple did ship some fixes to iconv to make it behave more like GNU iconv again ([example](https://github.com/apple-oss-distributions/libiconv/commit/3bb193e61b6c1ad159f20f9f39dfe55a691f3dd3#diff-3e17ad6feaa40f25e55086beae98edbfea8e74665faafc6a0e794e83c10d316fR90-R93)) but it seems to be a reactive change, so maybe there are more differences. It at least fixed some issues (e.g. Vim tests would fail in macOS 14.0). I can see why GNU prefers to just GNU iconv though as that would be what they test and develop with and it's hard for them to work around a different implementation. As I mentioned in https://github.com/Homebrew/homebrew-core/pull/142490#issuecomment-1752444627 I think macOS 13 shipped with GNU iconv which was why it never caused an issue before, until macOS 14 replaced it with a custom BSD licensed solution.

**Comment by Bo98** [comment]
To be clear, we were aware of the < 14.2 bug because we actually worked with Apple to get those fixed. The reports however stopped coming in after those 14.2 fixes landed and it was only when this update came out that we were even aware of the extra 14.4 bug.

**Comment by Bo98** [comment]
Given it was seemingly never done, I'll pass on the new bug to Apple and see if it will get looked into.

**Comment by chenrui333** [comment]
> Seems very messy if it bleeds into gettext.

gettext has the similar situation now, https://github.com/Homebrew/homebrew-core/pull/199639

**Pull request merged**